### DISCUSSION TITLE
[Python] What's the "Right Size" or "Right Grouping" of Concepts for a Concept Exercise?

### DISCUSSION THREAD

**Discussion by BethanyG** [exercism/v3#2115] [General] [answered]
As @DavidGerva and I go through the Python [concept list and tracking table](https://github.com/exercism/v3/tree/master/languages/python/reference) to log concept exercise _**issues**_, we're struggling a bit with concept "sizing". 

What smaller things (like `in` or `statements` or `pass` or `exiting loops`) belong together under a more general concept, and which merit a stand-alone exercise?  What should we explain in the context of something else? (_**e.g.**  `in` for membership testing being explained within each data structure that it can be called on._)

We've (sort have) been doing this grouping and editing after the fact by discussing each issue as it's logged and making adjustments -- but we'd like to maybe have some brainstorming or thoughts up front, so we don't waste effort logging and discussing something we turn around and close -- or at the very least we keep that sort of cycle to a minimum.

Are we going about this in the right way?  Should we be re-making our "tracking table" to show which things go together -- or do we just plow through with what we currently have listed, and close issues we deem "too small"?  Thoughts (_especially from Python maintainers_) welcome.

**Comment by SleeplessByte** [comment, marked as answer]
My non-python answer (from a non-python maintainer):

- if it's too easy to come up with _many_ idiomatic solutions, then it's too big
- if it feels like teaching multiple "things", then it's too big
- if you have two exercises that would need eachother as dependency, then it's too small

My suggestion: it's usually not _that_ hard to split it up. It's harder to bring them together after you've attempted to do the work twice.
